gensim/matutils.py

```diff
@@ -382,6 +383,93 @@ def cossim(vec1, vec2):
     return result
 
 
+def kullback_leibler(vec1,vec2,num_of_features=None):
+    """
+    A similarity metric between two probability distributions.
+    Returns a similarity in range <0,1> where values closer to 0 mean a higher similarity.
+    Uses the scipy.stats.entropy method to identify kullback_leibler convergence value.
+    If the distribution draws from a certain number of docs, that value must be passed.
+    """
+    if scipy.sparse.issparse(vec1) and scipy.sparse.issparse(vec2):
+        vec1 = vec1.todense()
+        vec2 = vec2.todense()
+        if vec1.shape[1] == 2 and vec2.shape[1] == 2:
+            vec1 = vec1.tolist()
+            vec2 = vec2.tolist()
+            max_len = max(len(vec1),len(vec2),num_of_features)
+            dense1 = sparse2full(vec1,max_len)
+            dense2 = sparse2full(vec2,max_len)     
+            return scipy.stats.entropy(dense1,dense2)
+        else:
+            return scipy.stats.entropy(vec1,vec2)
+    elif isinstance(vec1,numpy.ndarray) and isinstance(vec2,numpy.ndarray):
+        if vec1.shape[1] == 2 and vec2.shape[1] == 2:
+            vec1 = vec1.tolist()
+            vec2 = vec2.tolist()
+            max_len = max(len(vec1),len(vec2),num_of_features)
+            dense1 = sparse2full(vec1,max_len)
+            dense2 = sparse2full(vec2,max_len)
+            return scipy.stats.entropy(dense1,dense2)
+        else:
+            return scipy.stats.entropy(vec1,vec2)
+    elif type(vec1) is list and type(vec2) is list:
```

Comment:
Just use `else` (the "sequence of 2-tuples" is not necessarily a `list`; it can be an iterator, tuple...).